src/systems/systems/src/hydraulic/update_iterator.rs

```diff
@@ -0,0 +1,280 @@
+use crate::simulation::UpdateContext;
+use std::time::Duration;
+
+pub struct FixedStepLoop {
+    lag_time_accumulator: Duration,
+    time_step: Duration,
+    number_of_loops_this_iteration: u32,
+    number_of_loops_remaining: u32,
+}
+impl FixedStepLoop {
+    pub fn new(time_step: Duration) -> Self {
+        Self {
+            lag_time_accumulator: Duration::from_millis(0),
+            time_step,
+            number_of_loops_this_iteration: 0,
```

Comment:
**suggestion**: `number_of_loops_this_iteration` can be removed and all writes to it can write to `number_of_loops_remaining`.
```suggestion
```